### DISCUSSION TITLE
Is there a way to add error policy on a query?

### DISCUSSION THREAD

**Discussion by paschalidi** [urql-graphql/urql#2155] [Q&A] [answered]
In a project I am working on we have been migrating from apollo to urql. We also use contentful, which though is not so relevant.

In using their gql schema we figured that sometimes we might have an error looking like this ```"[GraphQL] Cannot query field "pageTitle" on type "LandingPage". Did you mean "cpageTitle"?"``` because of some change in the schema. 

I know in apollo you have the [`errorPolicy: 'all'`](https://www.apollographql.com/docs/react/data/error-handling/#setting-an-error-policy) option which will return both the error and the data in these cases.

I observed you also saying [here](https://formidable.com/open-source/urql/docs/basics/errors/) that returning both errors and the data is possible but I am not getting how to do that from reading the docs. Any advice?

**Comment by kitten** [comment, marked as answer]
Hiya,

`urql` always returns a `CombinedError` and `data`. However, in the case of a schema error your query won't run and no `data` will be returned 😅

**Reply by paschalidi** [reply]
Thx for the super quick response! 
I see what you are saying, so I guess I miss-understood the documentation. 

You reckon there is a way to filter out the fields that errored after a query has failed in such a way you describing?
